Client/Components/SidebarEditSection.razor

```diff
@@ -0,0 +1,46 @@
+<input
+    id="section-edit-input"
+    type="text"
+    class="form-control"
+    placeholder="Section Title"
+    @bind-value="@Title"
+    @bind-value:event="oninput"
+    @onblur="@OnSaveCallback"/>
+
+@code {
+
+    [Inject]
+    private IJSRuntime JsRuntime { get; set; }
+
+    [Parameter]
+    public long SectionId { get; set; }
+
+    [Parameter]
+    public string InitialTitle { get; set; }
+
+    [Parameter]
+    public Func<long, string, Task> OnSave { get; set; }
+
+    private string Title { get; set; }
+
+    private void OnSaveCallback()
+    {
+        OnSave(SectionId, Title);
+    }
+
+    protected override Task OnParametersSetAsync()
+    {
+        Title = InitialTitle;
+        return base.OnParametersSetAsync();
+    }
+
+    protected override void OnAfterRender(bool firstRender)
+    {
+        if (firstRender)

```

Comment:
what's this doing on first render? A comment here would help clear up any confusion

Reply:
it's using JS interop to call a JS function to focus the input element. I'll add a comment.